hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/TinkerpopBackend.java

```diff
@@ -113,7 +113,8 @@ public Element find(CanonicalPath element) throws ElementNotFoundException {
 
         q.counter("total").page(pager);
 
-        return new Page<>(q.cast(Element.class).toList(), pager, q.getCount("total"));
+        // here the toList is very expensive
+        return new Page<>(q.cast(Element.class).iterator(), pager, q.getCount("total"));
```

Comment:
I am not sure if the `q.getCount()` call will report the correct number without the `toList()`. Have you checked that?